### PR TITLE
[AMS-736] kafka event receiver splitted into pure receiver + reader

### PR DESCRIPTION
**Pull request by denisivan0v**

### CODE REVIEW

File: src/VStore/Kafka/CachingConsumerFactory.cs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                        consumer.OnLog += OnLog;
+                        consumer.OnError += OnError;
+                        consumer.OnStatistics += OnStatistics;
+                        consumer.OnConsumeError += OnConsumeError;
+
+                        return consumer;
+                    });
+        }
+
+        public Consumer<string, string> CreateConsumer(string brokerEndpoints, string groupId = null)
```
Comment (xakep139) [inline src/VStore/Kafka/CachingConsumerFactory.cs]: Second call to this method with different params will return the same consumer, so I think this colud lead to misunderstanding. Wouldn't it be better to pass this params in `ctor`?

Comment (denisivan0v) [reply src/VStore/Kafka/CachingConsumerFactory.cs]: Done in 802d47f

File: src/VStore.Worker/VStore.Worker.csproj [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
   <PropertyGroup>
     <TargetFramework>netcoreapp1.1</TargetFramework>
     <AssemblyName>VStore.Worker</AssemblyName>
     <RootNamespace>NuClear.VStore.Worker</RootNamespace>
     <OutputType>Exe</OutputType>
     <PackageId>VStore.Worker</PackageId>
-    <!--<RuntimeIdentifier>osx.10.12-x64</RuntimeIdentifier>-->
-    <RuntimeIdentifier>win10-x64</RuntimeIdentifier>
+    <RuntimeIdentifier>osx.10.12-x64</RuntimeIdentifier>
+    <!--<RuntimeIdentifier>win10-x64</RuntimeIdentifier>-->
```
Comment (xakep139) [inline src/VStore.Worker/VStore.Worker.csproj]: @denisivan0v don't forget to comment back before merging

File: src/VStore.Worker/Jobs/BinariesCleanupJob.cs [hunk 3]
```diff
@@ -122,14 +122,21 @@ private IDisposable CleanupBinaries(int batchSize, TimeSpan delay, CancellationT
                 var oldestSessionDate = sessionCreatingEvents[0].Timestamp.UtcDateTime;
                 var dateToStart = oldestSessionDate.Subtract(SafetyPeriod);
 
+                _logger.LogInformation(
+                    "Starting to process '{totalSessionsCount}' sessions. Oldest session date: {oldestSessionDate}. " +
+                    "Binary reference events stream will be readed starting from {dateToStart}.",
+                    sessionCreatingEvents.Count,
+                    oldestSessionDate.ToString("u"),
```
Comment (xakep139) [inline src/VStore.Worker/Jobs/BinariesCleanupJob.cs]: You can use formating in format string like `{dateToStart:u}`

**Comment by xakep139** [comment]
@denisivan0v, commit https://github.com/2gis/nuclear-vstore/pull/98/commits/a185762d6da7153ca8b12f89fd9c93917cb30699 is related to the article https://blog.markvincze.com/troubleshooting-high-memory-usage-with-asp-net-core-on-kubernetes/

**Comment by xakep139** [comment]
There are also issue https://github.com/aspnet/Home/issues/2056

**Pull request merged**